components/Sidebar.go

```diff
@@ -0,0 +1,230 @@
+package components
+
+import (
+	"github.com/gdamore/tcell/v2"
+	"github.com/rivo/tview"
+
+	"github.com/jorgerojas26/lazysql/app"
+	"github.com/jorgerojas26/lazysql/commands"
+	"github.com/jorgerojas26/lazysql/models"
+)
+
+type SidebarState struct {
+	currentFieldIndex int
+}
+
+type Sidebar struct {
+	*tview.Frame
+	Flex        *tview.Flex
+	state       *SidebarState
+	Fields      []*tview.TextArea
+	subscribers []chan models.StateChange
+}
+
+func NewSidebar() *Sidebar {
+	flex := tview.NewFlex().SetDirection(tview.FlexColumnCSS)
+	frame := tview.NewFrame(flex)
+	frame.SetBackgroundColor(tview.Styles.PrimitiveBackgroundColor)
+	frame.SetBorder(true)
+
+	sidebarState := &SidebarState{
+		currentFieldIndex: 0,
+	}
+
+	newSidebar := &Sidebar{
+		Frame:       frame,
+		Flex:        flex,
+		state:       sidebarState,
+		Fields:      []*tview.TextArea{},
+		subscribers: []chan models.StateChange{},
+	}
+
+	newSidebar.SetInputCapture(newSidebar.inputCapture)
+
+	newSidebar.SetBlurFunc(func() {
+		newSidebar.SetCurrentFieldIndex(0)
+	})
+
+	return newSidebar
+}
+
+func (sidebar *Sidebar) AddField(title, text string, fieldWidth int) {
+	field := tview.NewTextArea()
+	field.SetWrap(true)
+	field.SetDisabled(true)
+
+	field.SetBorder(true)
+	field.SetTitle(title)
+	field.SetTitleAlign(tview.AlignLeft)
+	field.SetTitleColor(tview.Styles.PrimaryTextColor)
+	field.SetText(text, true)
+	field.SetTextStyle(tcell.StyleDefault.Background(tview.Styles.PrimitiveBackgroundColor).Foreground(tview.Styles.SecondaryTextColor))
+
+	textLength := len(field.GetText())
+
+	itemFixedSize := 3
+
+	if textLength >= fieldWidth*3 {
+		itemFixedSize = 5
+	} else if textLength >= fieldWidth {
+		itemFixedSize = 4
+	} else {
+		field.SetWrap(false)
+	}
+
+	sidebar.Fields = append(sidebar.Fields, field)
+	sidebar.Flex.AddItem(field, itemFixedSize, 0, true)
+}
+
+func (sidebar *Sidebar) FocusNextField() {
+	newIndex := sidebar.GetCurrentFieldIndex() + 1
+
+	if newIndex < sidebar.Flex.GetItemCount() {
```

Comment:
Invert the test to simplify


```suggestion
	if newIndex >= sidebar.Flex.GetItemCount() {
		return
}
```